Bug: Settings::core() compilation error on main branch

## Prerequisites

- [x] I have searched existing issues to verify this bug hasn't been reported

## Bug Description

After merging PR #2861 (settings-fragment-accessor refactoring), `Cargo Check (bins-examples)` and `Cargo Check (tests)` CI jobs fail on the `main` branch with a compilation error:

```
error[E0599]: no method named `core` found for struct `settings::Settings` in the current scope
```

The error indicates that `Settings::core` is a **field, not a method**, but the code at `crates/reinhardt-conf/src/settings.rs:723` attempts to call it as `settings.core()`.

## Steps to Reproduce

1. Check out the `main` branch (commit after PR #2866 merge)
2. Run `cargo check --workspace --all --all-features`
3. Observe compilation error in `crates/reinhardt-conf/src/settings.rs:723`

## Expected Behavior

`cargo check --workspace --all --all-features` should succeed on the `main` branch.

## Actual Behavior

Compilation fails with:

```
error[E0599]: no method named `core` found for struct `settings::Settings` in the current scope
  --> crates/reinhardt-conf/src/settings.rs:723:23
   |
126 | pub struct Settings {
   | ------------------- method `core` not found for this struct
...
723 |         let core = settings.core();
   |                             ^^^^ field, not a method
   |
   ::: crates/reinhardt-conf/src/settings/core_settings.rs:117:5
```

## Environment

- Rust version: stable (CI)
- OS: ubuntu-latest (CI), macOS (local)
- Affected CI jobs: `Cargo Check (bins-examples)`, `Cargo Check (tests)`
- Workflow run: https://github.com/kent8192/reinhardt-web/actions/runs/23435574531

## Additional Context

- This error was introduced by PR #2861 (refactor/settings-fragment-accessor)
- The fix likely involves either:
  - Adding a `core()` accessor method to `Settings`, or
  - Changing `settings.core()` to direct field access `settings.core`
- This blocks all downstream CI jobs (Unit Tests, Integration Tests, etc.)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/crates/reinhardt-conf/src/settings.rs
+++ b/crates/reinhardt-conf/src/settings.rs
@@ -596,6 +596,7 @@ impl MiddlewareConfig {
 #[allow(deprecated)] // Tests exercise deprecated Settings for backward-compatibility verification
 mod tests {
 	use super::*;
+	use crate::settings::core_settings::HasCoreSettings;
 	use rstest::rstest;
 
 	#[rstest]

